### PR TITLE
upgrade to go1.17.6

### PR DESCRIPTION
**Pull request by djmitche**
### What does this PR do?

Upgrades Go

### Motivation

Go1.18 will be out shortly, and we prefer to be on the next-to-latest major revision.  go1.17.6 is the latest go1.17.

### QA Process

Nothing in the Go changelog suggests any particular class of issue.  Include tests on "unusual" architectures and deployment situations in the usual QA process.

As an example, in the last release, we uncovered a Go bug which caused `fork()` calls to hang on MacOS.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] The appropriate `team/..` label has been applied, if known.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by brycekahle — APPROVED** [review]

**Review by jtappa — APPROVED** [review]
👍🏻 from docs team

**Review by paulcacheux — APPROVED** [review]

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module github.com/DataDog/datadog-agent
 
-go 1.16
+go 1.17
```
Comment (mx-psi) [inline go.mod:3]: Needs a `inv -e tidy-all` I think

Comment (mx-psi) [reply go.mod:3]: I _hope_ we only need to do this once

**Comment by mx-psi** [comment]
```
go mod tidy: go.mod file indicates go 1.17, but maximum supported version is 1.16
```

File: go.mod (hunk 1 above)
Comment (djmitche) [reply go.mod:3]: The "for other options" link doesn't have anything about pruning or the `-compat` flag, so I'm sort of guessing.  But, it seems like being compat with 1.17 is OK :)

**Comment by djmitche** [comment]
Huh, looks like I missed a buildimage in https://github.com/DataDog/datadog-agent-buildimages/commit/6ff41c6b73f6435a2cc568beb71ff487c035cb60 because it doesn't use GIMME.  I'll update.

**Review by mx-psi — COMMENTED** [review]
You are missing bumping the CodeQL Go version (another new thing!). After that, we can merge this and merge the macOS Go bump immediately after

**Comment by djmitche** [comment]
For the build task:
```
Error response from daemon: Get https://registry-1.docker.io/v2/: Get https://auth.docker.io/token?account=packagedatadoghq&client_id=docker&offline_token=true&service=registry.docker.io: dial tcp: lookup auth.docker.io: Temporary failure in name resolution
```

**Comment by djmitche** [comment]
OK, looks good!  I'd like a look from agent-core, and any other teams are invited to get their ✅ in but I won't wait for them :)

**Review by sgnn7 — APPROVED** [review]

**Pull request merged**